### PR TITLE
Export Compliance ITC (#3998)

### PR DESCRIPTION
**Pull request by lacostej**
A first implementation

I am waiting for many export compliance results. And I can check it that way:

```
require 'fastlane_core'
UI = FastlaneCore::UI

user = ''

Spaceship.login(user)
Spaceship::Tunes.login(user)

all_apps = Spaceship::Tunes::Application.all

class CCatDownloader
  def self.download_ccat_file(apple_id, export_compliance)
    ccat_file = export_compliance.ccat_file
    result = self.download(ccat_file.url)
    file_name = "#{apple_id}_#{export_compliance.id}_#{ccat_file.name}"
    File.write(file_name, result)
    UI.success("Successfully downloaded CcatFile file - '#{file_name}'")    
  end

  def self.download(url)
    uri = URI.parse(url)
    http = Net::HTTP.new(uri.host, uri.port)
    http.use_ssl = (uri.scheme == "https")
    res = http.get(uri.request_uri)
    res.body
  end
end

all_apps.each { |app|
  begin
    ec = app.export_compliances
    UI.message("Checking statuses for #{app.apple_id} / #{app.name}")
    ec.each do |ec|
      extra_info = "builds: #{ec.builds} store_versions: #{ec.store_versions}"
      UI.message("ExportCompliance: status: #{ec.status} - code: #{ec.code_value} #{extra_info}")
      CCatDownloader.download_ccat_file(app.apple_id, ec)
    end
  rescue => ex
    UI.error "Error: #{ex.message}"
  end
}
```

### CODE REVIEW

**Comment by KrauseFx** [comment]
I don't know anything about export compliances, but this looks good to me. What things are missing?

**Comment by lacostej** [comment]
@KrauseFx there are a few fields in the json results that are not mapped yet.

https://github.com/fastlane/fastlane/pull/3999/files#diff-99fb0fa2e5efcaafd5d2ca35025a6d52R111

They are empty right now. Because my apps haven't been released yet. I don't know which structure they will hold. 

So we can either:
* ask someone to give an example of data so that I can write the missing bits
* wait for me to get the data and update the PR (~ next week I guess)
* release as is and update later

**Comment by lacostej** [comment]
@KrauseFx now that I know the format of the missing fields I updated the PR. From my end it's ready to review & merge.

File: spaceship/lib/spaceship/tunes/app_export_compliance.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      attr_reader :builds
+      attr_reader :store_versions
+
+      attr_mapping({
+        'id' => :id,
+        'uploadDate' => :upload_date,
+        'codeValue' => :code_value,
+        'platform' => :platform,
+        'status' => :status,
+        'exportComplianceRequired' => :export_compliance_compliance_required,
```
Comment (mfurtak) [inline spaceship/lib/spaceship/tunes/app_export_compliance.rb]: Just a quick double-check: Is the doubled "compliance" in `export_compliance_compliance_required` intended?

Comment (lacostej) [inline spaceship/lib/spaceship/tunes/app_export_compliance.rb]: oops !

Comment (lacostej) [inline spaceship/lib/spaceship/tunes/app_export_compliance.rb]: @mfurtak:I have updated the PR, fixing the above.

I just realized that there was some implementation of export compliance under app_submission: 
https://github.com/fastlane/fastlane/blob/7b5e166/spaceship/lib/spaceship/tunes/app_submission.rb#L82-L92

I could try refactoring a bit, or do it later. WDYT ?

**Comment by lacostej** [comment]
@mfurtak:I have updated the PR, fixing the original comment. 

Then hound code style has changed apparently. Needs fixup.

I just realized that there was some implementation of export compliance under app_submission: 
https://github.com/fastlane/fastlane/blob/7b5e166/spaceship/lib/spaceship/tunes/app_submission.rb#L82-L92

I could try refactoring a bit, or do it later. Any preference ?

**Comment by mfurtak** [comment]
Hm, if the existing code is dealing with the same data in a different place/way, I'd say it would be best not to introduce a second place/way. I'd say refactor first. 🙁

**Comment by KrauseFx** [comment]
What's the status on this PR?

**Comment by ohwutup** [comment]
Hey @lacostej! I am going to close this PR for now as it has been active for a while. But please feel free to reopen the PR if you'd like! Thanks! 🐙 👶

**Comment by lacostej** [comment]
@KrauseFx sorry I missed your ping in May. Status is here: https://github.com/fastlane/fastlane/pull/3999#issuecomment-214799365

I use the functionality as is. It helped me :)

I would prefer the PR to be kept open until I find the time to refactor it. It s just a bit difficult to allocate time on that issue right now.

**Comment by lacostej** [comment]
@ohwutup I don't seem to be able to reopen

**Comment by ohwutup** [comment]
Sorry about that @lacostej !

**Comment by KrauseFx** [comment]
Hey @lacostej, I'll close this pull request for now, just to get the number of open PRs down. Please re-open (or ping us to re-open), once you have time again to tackle refactoring the PR 👍  Thanks for all your contribution

**Pull request closed without merging**